src/runDiff.ts

```diff
@@ -0,0 +1,225 @@
+    const octokitPaginatedJobs = await this.octokit.paginate(
+      this.octokit.rest.actions.listJobsForWorkflowRun,
+      {
+        ...github.context.repo,
+        run_id: github.context.runId
+      }
+    );
+
+    const job = octokitPaginatedJobs.find(j => j.name === this.inputs.jobName);
+
+    if (job) {
+      return {
+        job_id: job.id,
+        html_url: job.html_url || ""
+      };
+    } else {
+      throw new Error(`Could not find job with name ${this.inputs.jobName}`);
+    }
+  }
+
+  /**
+   * Executes the CDKTF diff command and processes its output.
+   * Supports both real execution and test mode with stub output.
+   * 
+   * @returns Promise containing result code and summary
+   * @throws Error if the diff execution fails unexpectedly
+   */
+  async runDiff(): Promise<{ result_code: ActionOutputs["result_code"]; summary: string }> {
+    const outputPath = path.join(os.tmpdir() || "/tmp", "cdktf-diff.txt");
+    let diffCommand = this.inputs.stubOutputFile ? 
+      `cat ${this.inputs.stubOutputFile}` :
+      "CI=1 npx cdktf diff";
+
+    if (this.inputs.skipSynth) {
+      diffCommand += " --skip-synth";
+    }
+    diffCommand += ` ${this.inputs.stack}`;
+
+    try {
+      let output = "";
+      await exec.exec("bash", ["-c", diffCommand], {
+        listeners: {
+          stdout: (data: Buffer) => {
+            output += data.toString();
+          },
+          stderr: (data: Buffer) => {
+            output += data.toString();
+          }
+        },
+        cwd: this.inputs.workingDirectory
+      });
+
+      // Write output to file for parsing
+      fs.writeFileSync(outputPath, output);
+      
+      return this.parseOutput(output);
+    } catch (error) {
+      return { result_code: "1", summary: (error as Error).message };
+    }
+  }
+
+  /**
+   * Parses the output from CDKTF diff command and determines the result.
+   * 
+   * @param output - Raw output from the diff command
+   * @returns Object containing result code and summary
+   * @throws Error if the output cannot be parsed
+   */
+  private parseOutput(output: string): { result_code: ActionOutputs["result_code"]; summary: string } {
+    // eslint-disable-next-line no-control-regex
+    const cleanOutput = output.replace(/\x1B\[([0-9]{1,3}(;[0-9]{1,2})?)?[mGK]/g, "");
+
+    // Check for various output patterns and determine result
+    if (cleanOutput.includes("Planning failed. Terraform encountered an error")) {
+      const summary = cleanOutput.match(/Error: .*/)?.[0] || "Unknown error occurred";
+      return { result_code: "1", summary };
+    }
+
+    if (cleanOutput.includes("No changes. Your infrastructure matches the configuration.")) {
+      return { result_code: "0", summary: "No changes. Your infrastructure matches the configuration." };
+    }
+
+    const planMatch = cleanOutput.match(/Plan:.*/);
+    if (planMatch) {
+      return { result_code: "2", summary: planMatch[0] };
+    }
```

Comment:
Since we're creating these results ourselves, is there a reason we're using numeric returns instead of more-js type: `resultCode: 'success'`.

Reply:
Yes, these match the terraform exit codes that are returned from the plans.